### PR TITLE
gnome: document packaging icon themes

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

Document some details of packaging icon themes.

See also https://github.com/NixOS/nixpkgs/pull/85229#issuecomment-615911872.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: doc/languages-frameworks/gnome.xml [hunk 1]
```diff
@@ -40,6 +40,23 @@
    </para>
   </section>
 
+  <section xml:id="ssec-icon-theme-packaging">
+    <title>Packaging icon themes</title>
+
```
Comment (worldofpeace) [inline doc/languages-frameworks/gnome.xml:45]: I believe this whole chapter is about packaging, so why not expand what is above?

File: doc/languages-frameworks/gnome.xml [hunk 2]
```diff
@@ -40,6 +40,23 @@
...
+    <para>
+      Icon themes may inherit from other icon themes. The inheritance is specified using the `Inherits` key in the `index.theme` file distributed with the icon theme. According to the <link xlink:href="https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html">icon theme specification</link> icons not provided by the theme are looked for in its parent icon themes. Therefore the parent themes should be installed as dependencies for a more complete experience regarding the icon sets used.
```
Comment (jtojnar) [inline doc/languages-frameworks/gnome.xml:47]:
```suggestion
      Icon themes may inherit from other icon themes. The inheritance is specified using the <literal>Inherits</literal> key in the <filename>index.theme</filename> file distributed with the icon theme. According to the <link xlink:href="https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html">icon theme specification</link>, icons not provided by the theme are looked for in its parent icon themes. Therefore the parent themes should be installed as dependencies for a more complete experience regarding the icon sets used.
```

File: doc/languages-frameworks/gnome.xml [hunk 3]
```diff
@@ -40,6 +40,23 @@
...
+    </para>
+
+    <para>
+      The package <package>hicolor-icon-theme</package> provides a setup hook which makes symbolic links for the parent themes into the directory <filename>share/icons</filename> of the current theme directory in the nix store, making sure they can be found at runtime. For that to work the packages providing parent icon themes should be listed as propagated build dependencies, toghether with <package>hicolor-icon-theme</package>.
```
Comment (jtojnar) [inline doc/languages-frameworks/gnome.xml:51]:
```suggestion
      The package <package>hicolor-icon-theme</package> provides a setup hook which makes symbolic links for the parent themes into the directory <filename>share/icons</filename> of the current theme directory in the nix store, making sure they can be found at runtime. For that to work the packages providing parent icon themes should be listed as propagated build dependencies, together with <package>hicolor-icon-theme</package>.
```

File: doc/languages-frameworks/gnome.xml (hunk 1 above)
Comment (jtojnar) [reply doc/languages-frameworks/gnome.xml:45]: I think separate section is fine, since there is not much overlap between apps and themes.

**Comment by romildo** [comment]
Can this be merged?

**Pull request merged**